src/ophyd_async/testing/_assert.py

```diff
@@ -60,11 +61,19 @@ async def assert_reading(
 
     """
     actual_reading = await readable.read()
+    _assert_readings_approx_equal(expected_reading, actual_reading)
+
+
+def _assert_readings_approx_equal(expected, actual):
     approx_expected_reading = {
-        k: dict(v, value=approx_value(expected_reading[k]["value"]))
-        for k, v in expected_reading.items()
+        k: dict(
+            v,
+            value=approx_value(expected[k]["value"]),
+            timestamp=approx_value(expected[k]["timestamp"]),
```

Comment:
Would be nice to support not having to put timestamps in the dictionary
```suggestion
            timestamp=pytest.approx(expected[k].get("timestamp", ANY)),
```
Maybe alarm severity too?